autogen/agentchat/contrib/rag/vectordb.py

```diff
@@ -0,0 +1,199 @@
+from abc import ABC, abstractmethod
+from typing import List, Any, Callable
+from .datamodel import Document, Query, QueryResults, GetResults
+from .encoder import Encoder
+
+
+class VectorDB(ABC):
```

Comment:
We decided to use `Protocol` rather than `ABC` in the future. See: https://jellis18.github.io/post/2022-01-11-abc-vs-protocol/

Reply:
Done.